packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTableDetailsSteps.tsx

```diff
@@ -174,17 +205,34 @@ export const TransactionsTableDetailsSteps = ({
       {/* The final step, showing the destination chain */}
       <Step
         done={isTxCompleted(tx)}
-        text={`Funds arrived on ${destNetworkName}`}
+        failure={isTxExpired(tx) || isDestinationChainFailure}
+        text={
+          isTxExpired(tx) || isDestinationChainFailure
+            ? `Transaction ${
+                isDestinationChainFailure ? 'failed' : 'expired'
+              } on ${destinationNetworkName}`
+            : `Funds arrived on ${destinationNetworkName}`
+        }
         endItem={
-          destNetworkTxId && (
-            <ExternalLink
-              href={`${getExplorerUrl(destChainId)}/tx/${getDestNetworkTxId(
-                tx
-              )}`}
-            >
-              <ArrowTopRightOnSquareIcon height={12} />
-            </ExternalLink>
-          )
+          <>
+            {destinationNetworkTxId && (
+              <ExternalLink
+                href={`${getExplorerUrl(
+                  destinationChainId
+                )}/tx/${getDestinationNetworkTxId(tx)}`}
+              >
+                <ArrowTopRightOnSquareIcon height={12} />
+              </ExternalLink>
+            )}
+            {isDepositReadyToRedeem(tx) && (
+              <TransactionsTableRowAction
+                type={tx.isWithdrawal ? 'withdrawals' : 'deposits'}
+                isError={true}
+                tx={tx}
+                address={address}
+              />
+            )}
+          </>
```

Comment:
can we move it out as its own component and not pass it in as prop like this?
better for performance

Reply:
done